src/cyclic_queue.hpp

```diff
@@ -93,56 +87,43 @@ class CyclicQueue {
         return head[other];
     }
     /**
-     * @brief
-     * Enqueue
+     * @brief Enqueue
      *
      * @description
      * Enqueues the given value into the queue.
      * @param[in] T &value The value we wish to enqueue
-     * @return void
      */
     void enqueue(const T &value) {
         incrementTail();
         *tail = value;
     }
     /**
-     * @brief
-     * Pop
+     * @brief Pop
      *
      * @description
      * Pops the last added element from the container
      * @return T The last added value
      */
     T pop() {
-        if (count > 0) {
-            T return_value = *tail;
-            decrementTail();
-            return return_value;
-        } else {
-            return -1;
-        }
+        T return_value = *tail;
+        decrementTail();
+        return return_value;
```

Comment:
I see you've completely discarded the alternative flow. If you think it's unnecessary it's fine though.